conversation_simulator/src/StateMachine.js

```diff
@@ -226,4 +230,32 @@ export default class {
     }
     return split
   }
+
+  // Marks that a specific answer was already used on a certain state
+  markSolution (state, answer) {
+    let stateName = this.getStateName(state)
```

Comment:
pass in state name, decouple UserContext object from state machine

Reply:
Ok the name is now passed instead of the state object.